app/workers/proactive_project_sync_worker.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class ProactiveProjectSyncWorker
+  include Sidekiq::Worker
+  sidekiq_options queue: :small, lock: :until_executed
+
+  PLATFORMS = %w[NPM Maven].freeze
+  DEFAULT_LIMIT = 1000
+
+  def perform(limit = DEFAULT_LIMIT)
+    projects_query
+      .limit(limit)
+      .each(&:async_sync)
+  end
+
+  def projects_query
```

Comment:
Since this isn't called by anything other than this worker, it would be best to remove it from the public interface. It means the test for `perform` will have to do the heavy lifting of testing more cases, but it will make the worker's public interface less complex. I don't think it's worth moving this query to `Project` since it's pretty specific to this use case.